cloudify_hostpool/config/yaml_pool.py

```diff
@@ -76,7 +77,11 @@ def _create_host(_host):
                     "'host' or the 'ip_range' key")
 
     def _get_auth(self, host):
-        return host.get('auth') or self.default.get('auth')
+        auth = host.get('auth') or self.default.get('auth', {})
+        if auth.get('keyfile') and not os.access(auth['keyfile'], os.R_OK):
```

Comment:
the string 'keyfile' is repeated three times in this method.
its best just to do it once
keyfile = auth.get('keyfile) and use keyfile variable